tests/type-check/good/option.arr

```diff
@@ -0,0 +1 @@
+some(1).and-then(lam(x :: Number): x + 1 end)
```

Comment:
This confirms that `and-then` type-checks, but it doesn't confirm the return type.  Could you add a `num-opt :: Option<Number> = ...` binding, to confirm the return type as well?